spot_wrapper/calibration/calibration_util.py

```diff
@@ -0,0 +1,916 @@
+            cameras based on their positions in the `images` array. For example, index `0`
+            corresponds to the first camera, index `1` to the second camera, and so forth,
+            up to `n-1` (where `n` is the number of cameras).
+            If None, all cameras are registered to the first camera (index `0`). Defaults to None.
+
+            The relationship between `desired_stereo_pairs` and `images` is crucial. Each tuple in
+            `desired_stereo_pairs` should be in the form
+            `(origin_camera_idx, reference_camera_idx)`,
+            where `origin_camera_idx` and `reference_camera_idx` correspond to the
+            indices of the cameras in the `images` array. For each stereo pair,
+            the function attempts to calibrate the `origin_camera_idx`
+            relative to the `reference_camera_idx`. If no pairs are provided,
+            the function defaults to registering
+            all cameras to the first camera (index `0`).
+
+        charuco_board (cv2.aruco_CharucoBoard, optional): The Charuco board configuration used
+            for calibration. Defaults to SPOT_DEFAULT_CHARUCO.
+
+        aruco_dict (cv2.aruco_Dictionary, optional): The Aruco dictionary used
+            to detect the markers. Defaults to SPOT_DEFAULT_ARUCO_DICT.
+
+        camera_matrices (Optional[Dict], optional): A dictionary mapping camera indices
+            to their existing camera matrices. If a matrix is not provided for a camera,
+            it will be computed during calibration
+            if that camera is part of a stereo pair. Defaults to an empty dictionary.
+
+        dist_coeffs (Optional[Dict], optional): A dictionary mapping camera indices
+            to their distortion coefficients.
+            If coefficients are not provided for a camera, they will be computed during calibration.
+            Defaults to an empty dictionary.
+
+    Raises:
+        ValueError: Raised if fewer than two cameras are provided, as stereo calibration
+            requires at least two cameras.
+        ValueError: Raised if a stereo pair includes the same camera twice.
+        ValueError: Raised if a camera in a stereo pair cannot be calibrated.
+
+    Returns:
+        Dict: A dictionary containing the calibrated intrinsic and extrinsic parameters
+            for the specified stereo pairs.
+        The keys are formatted as "origin_camera_idx_reference_camera_idx",
+            and the values contain the calibration data.
+
+    """
+    calibration = {}
+    num_cams = images.shape[1]
+    if num_cams == 0 or num_cams == 1:
+        raise ValueError("Stereo Madness requires more than one camera. Use the monocular method instead.")
+    else:  # More than one camera - Multi-Stereo Madness ;p
+        for idx in range(num_cams):
+            try:
+                _ = camera_matrices[idx]
+            except KeyError:
+                camera_matrices[idx] = None
```

Comment:
I might prefer just checking `if idx in camera_matrices` rather than relying on a exception to throw. I'm not sure about the performance of catching exceptions in python, but it would probably be a bit clearer. 

```suggestion
            if idx not in camera_matrices:
                camera_matrices[idx] = None
```

Reply:
You're so right don't know why I didn't think of this at the time :sweat_smile: 

added that just like that